soc/xtensa/intel_adsp/tools/cavstool.py

```diff
@@ -659,30 +662,60 @@ async def _main(server):
         if dsp.HIPCIDA & 0x80000000:
             dsp.HIPCIDA = 1<<31 # must ACK any DONE interrupts that arrive!
 
+        if server:
+            # Check if the client connection is alive.
+            if not is_connection_alive(server):
+                lock.acquire()
+                start_output = False
+                lock.release()
+
 class adsp_request_handler(socketserver.BaseRequestHandler):
     """
     The request handler class for control the actions of server.
     """
 
-    def receive_fw(self, filename):
+    def receive_fw(self):
+        header_sz = 80
+        LARGE_BUF_SIZE = 16 * 1024 * 1024
```

Comment:
Looks like this is leftover from my bad `recvall()` suggestion (there is no `recvall` , sorry for misleading you). I think you can simplify and use 4K everywhere.